projects/rocprofiler-sdk/source/lib/python/rocpd/source/perfetto.cpp

```diff
@@ -225,6 +228,45 @@ write_perfetto(
                            std::unordered_map<rocprofiler_queue_id_t, ::perfetto::Track>>{};
     auto stream_tracks = std::unordered_map<rocprofiler_stream_id_t, ::perfetto::Track>{};
 
+    auto read_events = [&conn, &process](uint64_t event_id) {
+        return rocpd::read_sql_query<types::event>(
+            conn,
+            fmt::format(
+                "SELECT * FROM rocpd_event WHERE guid='{}' AND id={}", process.guid, event_id));
+    };
+
+    auto read_pmc_events = [&conn, &process](uint64_t event_id) {
+        return rocpd::read_sql_query<types::pmc_event>(
```

Comment:
```suggestion
    auto read_pmc_events = [&conn, &process, &ocfg](uint64_t event_id) {
        if(!ocfg.annotate_pmc) return std::vector<types::pmc_event>{};
        return rocpd::read_sql_query<types::pmc_event>(
```
This query is quite slow.  We should add the `--annotate-pmc` option and default it to false, so that this SQL query only runs IFF the parameter is specified to be true.... similar to Jonathan's branch.  

If you run `ctest -R rocpd` and compare the time taken to run the `tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation` test.  Compare WITH and WITHOUT `--annotate-pmc` on this KFD branch.

1) You'll need to make the `--annotate-pmc` changes from projects/rocprofiler-sdk/source/lib/output/output_config.hpp
2) Also need the `--annotate-pmc` changes in projects/rocprofiler-sdk/source/lib/python/rocpd/pftrace.py

That should do it... (or check Jonathan's branch/PR for any more annotate-pmc changes)

Reply:
Without kfd annotation: 3.03 sec
```
      Start 296: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation
13/37 Test #296: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation ........................   Passe
d    3.03 sec
```

With kfd annotation: 3.56 sec
```
      Start 296: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation
13/37 Test #296: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation ........................   Passed    3.28 sec
```

This test only generates about 10 KFD events, so small overhead is expected (and might even just be noise)

In my KFD stress test which generates thousands of KFD events in a short program, perfetto generation goes from ~1 second to ~3 seconds:
```
(venv) bkocolos@rocm-perf01:~/github/ROCm/hmmstress/hmmstress$ time ~/.local/rocm//bin/rocpd convert -i rocm-perf01/*.db --output-format pftrace -d rocm-perf01 -o trace --annotate-kfd
Found 1 database files.
Converting database(s) to pftrace format:
[325.407]       perfetto.cc:47304 Configured tracing session 1, #sources:1, duration:0 ms, #buffers:1, total buffer size:1048576 KB, total sessions:1, uid:0 session name: ""
W20251209 15:12:56.126623 135544034553984 simple_timer.cpp:55] Perfetto generation from SQL for process 50591 (write) ::     2.935205 sec
W20251209 15:12:56.128391 135544034553984 simple_timer.cpp:55] Perfetto generation from SQL for process 50591 (total) ::     2.979868 sec
W20251209 15:12:56.128424 135544034553984 simple_timer.cpp:55] Perfetto generation from 1 SQL database(s) ::     2.981177 sec
E20251209 15:12:56.129039 135544034553984 output_stream.cpp:111] Opened result file: rocm-perf01/trace_results.pftrace
Done. Exiting...
[328.448]       perfetto.cc:48888 Tracing session 1 ended, total sessions:0

real    0m3.221s
user    0m2.934s
sys     0m0.341s
(venv) bkocolos@rocm-perf01:~/github/ROCm/hmmstress/hmmstress$ time ~/.local/rocm//bin/rocpd convert -i rocm-perf01/*.db --output-format pftrace -d rocm-perf01 -o trace --no-annotate-kfd
Found 1 database files.
Converting database(s) to pftrace format:
[340.991]       perfetto.cc:47304 Configured tracing session 1, #sources:1, duration:0 ms, #buffers:1, total buffer size:1048576 KB, total sessions:1, uid:0 session name: ""
W20251209 15:13:09.687324 133611734737024 simple_timer.cpp:55] Perfetto generation from SQL for process 50591 (write) ::     0.910274 sec
W20251209 15:13:09.690749 133611734737024 simple_timer.cpp:55] Perfetto generation from SQL for process 50591 (total) ::     0.957377 sec
W20251209 15:13:09.690777 133611734737024 simple_timer.cpp:55] Perfetto generation from 1 SQL database(s) ::     0.959311 sec
E20251209 15:13:09.694407 133611734737024 output_stream.cpp:111] Opened result file: rocm-perf01/trace_results.pftrace
[341.998]       perfetto.cc:48888 Tracing session 1 ended, total sessions:0
Done. Exiting...

real    0m1.167s
user    0m0.894s
sys     0m0.317s

Reply:
Yup, while when --annotate-pmc was always run, I saw that test run approx 80s because that test has a lot of PMC data in it.

```
Start 572: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation
22/46 Test #572: tests.integration.execute.rocprofv3-test-rocpd-perfetto-generation ........................   Passed   86.08 sec
```